### PR TITLE
Do not hard set config in CPTC files

### PR DESCRIPTION
**Pull request by drashna**
## Description

This is needed or rules.mk changes get overriden. 

## Types of Changes

- [x] Core
- [x] Bugfix

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

File: platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk [hunk 1]
```diff
@@ -5,8 +5,8 @@ BOARD := QMK_PROTON_C
...
 # These are defaults based on what has been implemented for ARM boards
-AUDIO_ENABLE = yes
-WS2812_DRIVER = bitbang
+AUDIO_ENABLE ?= yes
+WS2812_DRIVER ?= bitbang
 
 # Force task driven PWM until ARM can provide automatic configuration
-BACKLIGHT_DRIVER = software
\ No newline at end of file
+BACKLIGHT_DRIVER ?= software
```
Comment (zvecr) [inline platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk:12]: This breaks converting an avr board that is currently using PWM backlight. The reason this sets software is as it needs zero config.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk (hunk 1 above)
Comment (drashna) [reply platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk:12]: In this case, wouldn't it be better to set the backlight driver defaults in `common_features.mk` based on the platform, then? 

Something like: 
```make
ifeq ($(PLATFORM),CHIBIOS)
    BACKLIGHT_DRIVER ?= software
else
    BACKLIGHT_DRIVER ?= pwm
endif
```

Comment (zvecr) [reply platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk:12]: The downside there, is it bleeds the CTPC implementation out. Right now, while maybe not the perfect behaviour, is at least isolated.

Comment (drashna) [reply platforms/chibios/QMK_PROTON_C/convert_to_proton_c.mk:12]: In that case, I add this change then.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**